### PR TITLE
Fixing #445

### PR DESCRIPTION
**Pull request by djibux**
Hopefully fixing #445

### CODE REVIEW

**Comment by mpcjanssen** [comment]
Thanks, I am a bit busy at the moment. Will have a look somewhere next week I hope.

**Comment by smichel17** [comment]
The logic looks sound to me. Slight optimization:

```
if (filter.createIsThreshold) {
    return thresholdDate?:createDate?:empty
} else {
    return thresholdDate?:empty
}
```
to `return thresholdDate?:filter.createIsThreshold?createDate:empty`

File: src/main/java/nl/mpcjanssen/simpletask/task/Task.kt [hunk 1]
```diff
@@ -298,7 +299,7 @@ class Task(text: String, defaultPrependedDate: String? = null) {
         }.map {it.text}.joinToString(" ")
     }
 
-    fun getHeader(sort: String, empty: String): String {
+    fun getHeader(sort: String, empty: String, filter: ActiveFilter): String {
```
Comment (smichel17) [inline src/main/java/nl/mpcjanssen/simpletask/task/Task.kt]: `fun getHeader(sort: String, empty: String, createIsThreshold: Boolean): String {`

File: src/main/java/nl/mpcjanssen/simpletask/task/Task.kt [hunk 2]
```diff
@@ -312,7 +313,11 @@ class Task(text: String, defaultPrependedDate: String? = null) {
                 return empty
             }
         } else if (sort.contains("by_threshold_date")) {
-            return thresholdDate?:empty
```
Comment (smichel17) [inline src/main/java/nl/mpcjanssen/simpletask/task/Task.kt]: `return thresholdDate?:createIsThreshold?createDate:empty`

File: src/main/java/nl/mpcjanssen/simpletask/util/Util.kt [hunk 3]
```diff
@@ -146,7 +146,7 @@ fun addHeaderLines(visibleTasks: List<TodoItem>, filter: ActiveFilter, no_header
     var headerLine: HeaderLine? = null
     for (item in visibleTasks) {
         val t = item.task
-        val newHeader  = LuaInterpreter.onGroupCallback(filter.moduleName,t) ?: t.getHeader(firstSort, no_header)
+        val newHeader  = LuaInterpreter.onGroupCallback(filter.moduleName,t) ?: t.getHeader(firstSort, no_header, filter)
```
Comment (smichel17) [inline src/main/java/nl/mpcjanssen/simpletask/util/Util.kt]: `val newHeader  = LuaInterpreter.onGroupCallback(filter.moduleName,t) ?: t.getHeader(firstSort, no_header, filter.createIsThreshold)`

File: src/main/java/nl/mpcjanssen/simpletask/task/Task.kt [hunk 4]
```diff
@@ -26,6 +26,7 @@
  */
 package nl.mpcjanssen.simpletask.task
 
+import nl.mpcjanssen.simpletask.ActiveFilter
```
Comment (smichel17) [inline src/main/java/nl/mpcjanssen/simpletask/task/Task.kt]: And then you can remove this.

**Comment by djibux** [comment]
Cool.
I'm not too familiar with the syntax yet :)

Le 26 août 2016 21:32:09 GMT+02:00, smichel17 <notifications@github.com> a écrit :
>The logic looks sound to me. Slight optimization:
>
>```
>if (filter.createIsThreshold) {
>    return thresholdDate?:createDate?:empty
>} else {
>    return thresholdDate?:empty
>}
>```
>to `return thresholdDate?:filter.createIsThreshold?createDate:empty`
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/mpcjanssen/simpletask-android/pull/540#issuecomment-242830126

-- 
djib

**Comment by smichel17** [comment]
@djibux haha, me neither; I had to look it up :P

**Pull request merged**